### PR TITLE
Assignment 1 import fixed, test worked w/o changes

### PR DESCRIPTION
**Pull request by archwu**
Hi Ron,

About the tests you got errors, if you run `PYTHONPATH=inventory_management/ python -m unittest test_unit` and it would work perfect. This PYTHONPATH is the solution of Bryan on the cannot find module errors. Below is the link.
https://gist.github.com/brydavis/9d3ace8f980fba955a145e963677327e

Still working on fixes of assignment 2

### CODE REVIEW

**Comment by archwu** [comment]
assignment 2 fixed, I recalled that I modified the source.json when I was debugging(deleted the extra ","). Is this the way that I should do it? One further question on this, if I have to load a buggy input json, can I do something to automatically skip the corrupted part of data and load the good one without crashing and quit loading?

**Comment by nairraghav** [comment]
> assignment 2 fixed, I recalled that I modified the source.json when I was debugging(deleted the extra ","). Is this the way that I should do it? One further question on this, if I have to load a buggy input json, can I do something to automatically skip the corrupted part of data and load the good one without crashing and quit loading?

For this assignment, that is fine. In general, when using the JSON Decoder (within the json library), it will only throw a Decode Error when you have an invalid JSON file. As far as I am aware, there is no way to "skip" that specific entry and move on (there may an external library for this or you could write your own 😉)

File: students/ArchWu/lesson03/assignment/src/basic_operations.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            )
+            customer.save()
+            logging.info("added customer %s", customer_id)
+
+    except Exception as e:
+        logging.error("Error adding customer %s", customer_id)
+
+def delete_customer(customer_id):
+    """Delete a customer of given customer_id"""
+    Customer.delete().where(Customer.customer_id == customer_id).execute()
```
Comment (nairraghav) [inline students/ArchWu/lesson03/assignment/src/basic_operations.py]: It is a good idea to determine if this user exists before trying to run the delete. That way you can throw an error and let the User know that this user does not exist

File: students/ArchWu/lesson03/assignment/src/basic_operations.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            'credit_limit':customer.credit_limit
+        }
+    except Exception as e:
+        logging.error("Error when finding customer %s due to %s", customer_id, e)
+        return {}
+
+def update_customer_credit(customer_id, credit_limit):
+    """Update the credit_limit of customer of given customer_id"""
+    try:
+        customer = Customer.update(credit_limit=credit_limit).where(Customer.customer_id == customer_id).execute()
```
Comment (nairraghav) [inline students/ArchWu/lesson03/assignment/src/basic_operations.py]: Similar to the delete, find the customer first, note if they exist, and then make the change

**Comment by nairraghav** [comment]
Nice work Arch, take a look at some of the comments I have left about how to improve the Customer queries a little.

**Comment by archwu** [comment]
@nairraghav 
Hi Ron,
I just fixed the code you mentioned above, however, I persistently get the error:
`self = <ExceptionInfo TypeError tblen=2>

    @property
    def value(self):
        """the exception value"""
>       return self._excinfo[1]
E       TypeError: 'NoneType' object is not subscriptable

/home/arch/anaconda3/lib/python3.7/site-packages/_pytest/_code/code.py:445: TypeError`
and I have no idea what cause this or how to solve it, do you have any suggestions?

Thanks:)

**Pull request merged**